spec/visual-regression-tester.js

```diff
@@ -32,13 +32,24 @@ class VisualRegressionTester {
     const { Page, Emulation } = cdp;
     const metrics = clone(this.metrics);
 
-    metrics.fitWindow = true;
-
     return Emulation.setDeviceMetricsOverride(metrics)
       .then(() => Page.getLayoutMetrics())
       .then((result) => {
-        metrics.height = result.contentSize.height;
-        return Emulation.setDeviceMetricsOverride(metrics);
+        // This is weird, I'm not really sure what I'm doing, but after
+        // a bunch of experimentation, this seems to do what we want, i.e.
+        // capture a full-size screenshot of the entire page.
+        //
+        // At the time of this writing, though, Emulation.setVisibleSize()
+        // is documented as being both experimental *and* deprecated, so
+        // this code may not work for long. - AV 7/28/2017
+        return Emulation.setVisibleSize({
```

Comment:
I think the most successful thing would be to take a screenshot of each component in isolation. That would mean if you change the height of something at the top of the page, it wouldn't invalidate everything below it. That being said, I think that's out of scope for this work.